lib/book/book_page/new_book_page.dart

```diff
@@ -0,0 +1,305 @@
+import 'dart:io';
+import 'dart:ui' as ui;
+
+import 'package:book/app_colors.dart';
+import 'package:book/app_routes.dart';
+import 'package:book/book/book_chapter/book_chapter.dart';
+import 'package:book/book/book_page/book_page.dart';
+import 'package:book/book/book_page/book_page_image.dart';
+import 'package:book/core/constants.dart';
+import 'package:book/enums/book_page_mode.dart';
+import 'package:book/login/widgets/custom_text_form_field.dart';
+import 'package:book/utils/dialog_utils.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+import 'package:image_picker/image_picker.dart';
+
+import 'bloc/book_page_bloc.dart';
+import 'bloc/book_page_event.dart';
+import 'bloc/book_page_state.dart';
+
+class NewBookPage extends StatefulWidget {
+  NewBookPage({
+    required this.newPage,
+    required this.bookId,
+    required this.chapters,
+    required this.pageMode,
+  });
+
+  BookPage newPage;
+  String bookId;
+  List<BookChapter> chapters;
+  BookPageMode pageMode;
+
+  @override
+  State<NewBookPage> createState() => _NewBookPageState();
+}
+
+class _NewBookPageState extends State<NewBookPage> {
+  final _formKey = GlobalKey<FormState>();
+  late String textValue;
+  late ImagePicker _picker;
+  File? image;
+  ui.Image? decodedImage;
+  BookChapter? selectedChapter;
+
+  @override
+  void initState() {
+    super.initState();
+    textValue = widget.newPage.text;
+    _picker = ImagePicker();
+
+    if (widget.newPage.pageImage != null) {
+      image = widget.newPage.pageImage!.image;
+    }
+    if (widget.pageMode == BookPageMode.edit) {
+      selectedChapter = widget.newPage.chapter;
+    } else {
+      selectedChapter = widget.chapters.lastOrNull;
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return BlocConsumer<BookPageBloc, BookPageState>(
+      listener: (context, state) async {
+        if (state is LoadingState) {
+          DialogUtils.showLoadingScreen(context);
+        } else if (state is LoadedState) {
+          Navigator.pop(context);
+        } else if (state is ErrorState) {
+          final snackBar = SnackBar(
+            content: Text(AppLocalizations.of(context)!.serverError),
+            backgroundColor: AppColors.errorSnackBar,
+          );
+          ScaffoldMessenger.of(context).showSnackBar(snackBar);
```

Comment:
make some snackbar utils. with success and error actions and forward the message only